server/models/ClientRequest.js

```diff
@@ -183,26 +184,32 @@ class ClientRequest {
     });
   }
 
-  checkHash(options) {
-    const hashes = this.getEnsuredArray(options.hashes);
-    const stoppedHashes = hashes.filter(hash => {
-      return torrentService.getTorrent(hash).status.includes(torrentStatusMap.stopped);
-    });
-    const hashesToStart = [];
+  checkHash(userId, options) {
+    const torrentService = ServicesHandler.getTorrentService(userId);
+    torrentService.fetchTorrentList().then(
+      () => {
+        const hashes = this.getEnsuredArray(options.hashes);
+        const stoppedHashes = hashes.filter(hash => {
+          return torrentService.getTorrent(hash).status.includes(torrentStatusMap.stopped);
+        });
 
-    this.stopTorrents({ hashes });
+        const hashesToStart = [];
 
-    hashes.forEach(hash => {
-      this.requests.push(this.getMethodCall('d.check_hash', [hash]));
+        this.stopTorrents({hashes});
 
-      if (!stoppedHashes.includes(hash)) {
-        hashesToStart.push(hash);
-      }
-    });
+        hashes.forEach(hash => {
+          this.requests.push(this.getMethodCall('d.check_hash', [hash]));
 
-    if (hashesToStart.length) {
-      this.startTorrents({ hashes: hashesToStart });
-    }
+          if (!stoppedHashes.includes(hash)) {
+            hashesToStart.push(hash);
+          }
+        });
+
+        if (hashesToStart.length) {
+          this.startTorrents({hashes: hashesToStart});
+        }
+      }
+    ).catch((err) => console.log(err));
```

Comment:
Let's kill the `console.log`. Also we're not actually handling the error — this method would have thrown previously.